### PR TITLE
UP 90 & 93

### PR DESCRIPTION
**Pull request by fontainesw**
Included are the changes for up 90 and 93. UP-93 has no implementation, as there are no methods that can  be called for quantity values

### CODE REVIEW

**Review by Lou3797 — APPROVED** [review]

**Review by Ryan-Guinn — APPROVED** [review]

File: packages/app/pages/Api/unit.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+export const Unit = (props) => {
+  return(
+    <div>
+      <div>id: {props.id}</div>
+      <div>name: {props.name}</div>
+      <div>symbol: {props.symbol}</div>
+      <br/>
+    </div>
+  )
```
Comment (ConnorHibbs) [inline packages/app/pages/Api/unit.tsx]: Statements should have a semicolon after them. return(); Checking tslint

File: packages/app/pages/Api/unit.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+const UnitField = (props) =>{
+  return(
+    <div>
+      <form onSubmit={props.onSubmitAction}>
+        Enter an Id: <input type="text" name="value" onChange={props.setUnit}/>
+        <input type="submit" value="query"/>
+      </form>
+    </div>
+  )
```
Comment (ConnorHibbs) [inline packages/app/pages/Api/unit.tsx]: ;

File: packages/app/pages/Api/unit.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    loading ? <strong>Loading...</strong> : (
+      error ? <p style={{color: "#F00"}}>API error</p> : (
+        <div>
+          <div>
+            {unitList.map( (unit) => (<Unit key={unit.id} id={unit.id} name={unit.name} symbol={unit.symbol}/>))}
+          </div>
+        </div>
+      )
+    )
+  );
```
Comment (ConnorHibbs) [inline packages/app/pages/Api/unit.tsx:48]: ;

File: packages/app/pages/Api/unit.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        <h2>List of all units: </h2>
+        <br/>
+        <UnitList/>
+        <br/>
+        <h2>Get Unit by Id: </h2>
+        <br/>
+        <UnitField setUnit={this.getUnitById} onSubmitAction={this.stopRefresh}/>
+        {getOneUnitId ? <GetSingleUnit unitId={getOneUnitId}/> : <div>Enter a value</div>}
+      </div>
+    )
```
Comment (ConnorHibbs) [inline packages/app/pages/Api/unit.tsx]: ;

File: packages/app/pages/Api/unit.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+  getUnitById = (event) => {
+    this.setState({setOneUnitId: parseInt(event.target.value)});
+  };
+
+  //Runs when "submit" is selected
+  stopRefresh = (event) => {
+    //Sets the value to query to the current value of the input field
+    this.setState({getOneUnitId: this.state.setOneUnitId});
+    event.preventDefault();
+  }
```
Comment (ConnorHibbs) [inline packages/app/pages/Api/unit.tsx]: ;

File: packages/app/pages/Api/unit.tsx (hunk 4 above)
Comment (ConnorHibbs) [inline packages/app/pages/Api/unit.tsx]: ;

File: packages/app/tests/Api/Unit-Test.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import tape from 'tape';
+import addAssertions from 'extend-tape';
+import jsxEquals from 'tape-jsx-equals';
+const test = addAssertions(tape, {jsxEquals});
+
+// Component to test
+import {Unit, UnitList, GetSingleUnit}  from '../../pages/Api/unit';
+
+
+test('UnitDisplaysUnit', (t) => {
```
Comment (ConnorHibbs) [inline packages/app/tests/Api/Unit-Test.js]: t might not be the best variable for readability. It is not clear what it stands for, without knowing what the "test" function takes as parameters

File: packages/ui-bindings/QuantityValue/quantityValue.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import { gql } from 'react-apollo'
+import {Unit} from "../Unit/unit";
+
+export const quantityValueInterface = gql`
+fragment quantityValueInterface on QuantityValue {
+  numericValue
+  unit{
+    id
+  }
+}`
```
Comment (ConnorHibbs) [inline packages/ui-bindings/QuantityValue/quantityValue.tsx]: End all statements with a semicolon (according to the linter for the project)

File: packages/ui-bindings/Unit/getUnitById.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  // bind input data from the store
+  connect((state: AppState) => ({
+    variables: {
+      token: getActiveLoginToken(state),
+    },
+  })),
+
+  graphql(query, {
+    // read query vars into query from input data above
+    options: (props) => (
```
Comment (ConnorHibbs) [inline packages/ui-bindings/Unit/getUnitById.tsx:38]: None of these statements are aligned. It makes it more difficult to read what is embedded in what. Please consider aligning. In JetBrains IDEs, CTRL + A, CTRL + ALT + i

File: packages/ui-bindings/Unit/unit.tsx [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+ */
+
+import { gql } from 'react-apollo'
+
+export const unitInterface = gql`
+fragment unitInterface on Unit {
+  id
+  name
+  symbol
+}`
```
Comment (ConnorHibbs) [inline packages/ui-bindings/Unit/unit.tsx]: statements end in semicolons according to the linter

File: packages/ui-bindings/Unit/getAllUnits.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+const query = gql`
+query($token: String) {
+  viewer(token: $token) {
+    allUnits{
+      ...unitInterface
+    }
+  }
+}
+${unitInterface}
+`
```
Comment (ConnorHibbs) [inline packages/ui-bindings/Unit/getAllUnits.tsx]: ;

File: packages/ui-bindings/Unit/getAllUnits.tsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
+/**
+ * A method to get all Units
+ *
+ * @package: REA app
+ * @author:  Steven Fontaine <fontainesw@msoe.edu>
+ * @since:   2017-12-04
+ */
+
+import { connect } from 'react-redux'
```
Comment (ConnorHibbs) [inline packages/ui-bindings/Unit/getAllUnits.tsx]: imports get semicolons

File: packages/ui-bindings/Unit/getAllUnits.tsx [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+
+export default compose(
+  // bind input data from the store
+  connect((state: AppState) => ({
+    variables: {
+      token: getActiveLoginToken(state),
+    },
+  })),
+
+  graphql(query, {
```
Comment (ConnorHibbs) [inline packages/ui-bindings/Unit/getAllUnits.tsx:36]: Very nice formatting in this compose. :+1:

File: packages/ui-bindings/QuantityValue/quantityValue.tsx [hunk 13]
```diff
@@ -0,0 +1,23 @@
+/**
+ * A This file defines common interfaces for the Unit object
+ *
+ * @package: REA app
+ * @author:  Steven Fontaine <fontainesw@msoe.edu>
+ * @since:   2017-12-04
+ */
+
+import { gql } from 'react-apollo'
```
Comment (ConnorHibbs) [inline packages/ui-bindings/QuantityValue/quantityValue.tsx]: imports get semicolons

File: packages/app/tests/Api/Unit-Test.js (hunk 6 above)
Comment (fontainesw) [reply packages/app/tests/Api/Unit-Test.js]: That is the generic for the function as described in the tape docs

**Comment by ConnorHibbs** [comment]
Overall, code structure looks good. There is one fix for variable naming. Please also add some more explanation around the testing stuff explaining what it is doing.

In every file, there were a lot of linter problems. If we need to revise the linter, we can have that conversation and propose a new coding style. But from the one we set in Fall quarter, there are numerous things that need to be fixed. CTRL + ALT + ENTER over any linter error should give you the option to fix the entire file (in JetBrains products) so that shouldn't be too difficult. 

Please also check formatting and alignment. In some of the more complex structures (Apollo's compose() function) the formatting is all over the place and it is very difficult to read.

Nice job getting the first PR this term!

**Review by ConnorHibbs — CHANGES_REQUESTED** [review]
Overall, code structure looks good. There is one fix for variable naming. Please also add some more explanation around the testing stuff explaining what it is doing.

In every file, there were a lot of linter problems. If we need to revise the linter, we can have that conversation and propose a new coding style. But from the one we set in Fall quarter, there are numerous things that need to be fixed. CTRL + ALT + ENTER over any linter error should give you the option to fix the entire file (in JetBrains products) so that shouldn't be too difficult.

Please also check formatting and alignment. In some of the more complex structures (Apollo's compose() function) the formatting is all over the place and it is very difficult to read.

Nice job getting the first PR this term!

File: packages/app/pages/Api/unit.tsx (hunk 3 above)
Comment (fontainesw) [reply packages/app/pages/Api/unit.tsx:48]: Resolved

Comment (fontainesw) [reply packages/app/pages/Api/unit.tsx:48]: resolved

File: packages/ui-bindings/Unit/getAllUnits.tsx (hunk 12 above)
Comment (fontainesw) [reply packages/ui-bindings/Unit/getAllUnits.tsx:36]: *very quiet I told you so*

File: packages/ui-bindings/Unit/getUnitById.tsx (hunk 8 above)
Comment (fontainesw) [reply packages/ui-bindings/Unit/getUnitById.tsx:38]: resolved

File: packages/ui-bindings/Unit/getAllUnits.tsx (hunk 12 above)
Comment (ConnorHibbs) [reply packages/ui-bindings/Unit/getAllUnits.tsx:36]: That's fair. I definitely deserve this

**Review by ConnorHibbs — APPROVED** [review]
Thanks for fixing the formatting!

**Pull request merged**